lib/currency.ex

```diff
@@ -92,15 +103,12 @@ defmodule Monetized.Currency do
 
   """
 
-  def parse_by_symbol(str) do
-    x = Enum.map(all, fn {k, v} -> {v.symbol, k} end) |> Enum.into(%{})
-    case Regex.run(~r/\p{Sc}/u, str) do
-      [s] ->
-        get(x[s])
-      _ ->
-        nil
-    end
-  end
+  @currencies
+  |> Enum.map(&Map.to_list/1)
+  |> Enum.each(fn (currency) ->
+    def parse_by_symbol(unquote(Keyword.get(currency, :symbol)) <> _rest), do: Enum.into(unquote(currency), %{})
```

Comment:
This matches by any string that starts w/ the symbol, so if needing to support currency symbol parsing anywhere else, then the regex approach would be better suited. Could at least add the regex parsing as a fallback.